### PR TITLE
Color Overlay Opacity

### PR DESCRIPTION
**Pull request by gregmax17**

### CODE REVIEW

File: filters/color-overlay/src/ColorOverlayFilter.ts [hunk 1]
```diff
@@ -27,15 +27,19 @@ type Color = number | number[] | Float32Array;
...
     /**
      * @param {number|Array<number>} [color=0x000000] - The resulting color, as a 3 component RGB e.g. [1.0, 0.5, 1.0]
+     * @param {number} [alpha=0] - The alpha value of the color
      */
-    constructor(color: Color = 0x000000)
+    constructor(color: Color = 0x000000, alpha: number)
     {
         super(vertex, fragment);
         this.uniforms.color = new Float32Array(3);
+        this.uniforms.alpha = alpha;
```
Comment (bigtimebuddy) [inline filters/color-overlay/src/ColorOverlayFilter.ts:40]: You don't need to set this here, The uniforms is set by setting `alpha` below.

File: filters/color-overlay/src/ColorOverlayFilter.ts [hunk 2]
```diff
@@ -27,15 +27,19 @@ type Color = number | number[] | Float32Array;
...
 {
     private _color = 0x0;
+    private _alpha = 0;
 
     /**
      * @param {number|Array<number>} [color=0x000000] - The resulting color, as a 3 component RGB e.g. [1.0, 0.5, 1.0]
+     * @param {number} [alpha=0] - The alpha value of the color
      */
-    constructor(color: Color = 0x000000)
+    constructor(color: Color = 0x000000, alpha: number)
```
Comment (bigtimebuddy) [inline filters/color-overlay/src/ColorOverlayFilter.ts:36]: This is marked as optional, but the typing does not indicate that. Would suggest setting the default value here `alpha:number = 1`

File: filters/color-overlay/src/ColorOverlayFilter.ts [hunk 3]
```diff
@@ -27,15 +27,19 @@ type Color = number | number[] | Float32Array;
 class ColorOverlayFilter extends Filter
 {
     private _color = 0x0;
+    private _alpha = 0;
 
     /**
      * @param {number|Array<number>} [color=0x000000] - The resulting color, as a 3 component RGB e.g. [1.0, 0.5, 1.0]
+     * @param {number} [alpha=0] - The alpha value of the color
```
Comment (bigtimebuddy) [inline filters/color-overlay/src/ColorOverlayFilter.ts:34]: The default should be `1`, no? I'd assuming setting `0` would make the color overlay invisible. To not be breaking change, this should be fully visible.

File: filters/color-overlay/src/ColorOverlayFilter.ts [hunk 4]
```diff
@@ -64,6 +68,19 @@ class ColorOverlayFilter extends Filter
     {
         return this._color;
     }
+
+    /**
+     * The alpha value of the color
+     * @member {number}
+     * @default 0
+     */
+    set alpha(value: number) {
```
Comment (bigtimebuddy) [inline filters/color-overlay/src/ColorOverlayFilter.ts:77]: Run `npm run lint -- --fix` to resolve some of these issues.

File: tools/demo/src/filters/color-overlay.js [hunk 5]
```diff
@@ -2,10 +2,11 @@ export default function ()
 {
     this.addFilter('ColorOverlayFilter', {
         fishOnly: true,
-        args: [0xff0000],
+        args: [0xff0000, 0.5],
```
Comment (bigtimebuddy) [inline tools/demo/src/filters/color-overlay.js:5]: preference to default to `1` not `0.5` in the demo

**Review by bigtimebuddy — CHANGES_REQUESTED** [review]
Looks good some minor stuff

File: filters/color-overlay/src/ColorOverlayFilter.ts (hunk 4 above)
Comment (gregmax17) [reply filters/color-overlay/src/ColorOverlayFilter.ts]: ok, I ran it

File: tools/demo/src/filters/color-overlay.js (hunk 5 above)
Comment (gregmax17) [reply tools/demo/src/filters/color-overlay.js]: set it to `1`

File: filters/color-overlay/src/ColorOverlayFilter.ts (hunk 3 above)
Comment (gregmax17) [reply filters/color-overlay/src/ColorOverlayFilter.ts:34]: set it to `1`

File: filters/color-overlay/src/ColorOverlayFilter.ts (hunk 1 above)
Comment (gregmax17) [reply filters/color-overlay/src/ColorOverlayFilter.ts]: doh, you're right. removed

File: filters/color-overlay/src/ColorOverlayFilter.ts (hunk 2 above)
Comment (gregmax17) [reply filters/color-overlay/src/ColorOverlayFilter.ts]: set it to `1`

**Review by bigtimebuddy — APPROVED** [review]
Great work thanks!

File: filters/color-overlay/src/ColorOverlayFilter.ts [hunk 6]
```diff
@@ -27,15 +27,18 @@ type Color = number | number[] | Float32Array;
 class ColorOverlayFilter extends Filter
 {
     private _color = 0x0;
+    private _alpha = 1;
 
     /**
      * @param {number|Array<number>} [color=0x000000] - The resulting color, as a 3 component RGB e.g. [1.0, 0.5, 1.0]
+     * @param {number} [alpha=0] - The alpha value of the color
```
Comment (bigtimebuddy) [inline filters/color-overlay/src/ColorOverlayFilter.ts:34]:
```suggestion
     * @param {number} [alpha=1] - The alpha value of the color
```

**Review by bigtimebuddy — APPROVED** [review]

**Pull request merged**